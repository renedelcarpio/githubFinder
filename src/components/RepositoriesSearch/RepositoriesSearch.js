import { useState } from 'react';
import { Repo } from '../../services/services';
import ErrorMessage from '../ErrorMessage/ErrorMessage';
import Form from '../Form/Form';
import Loader from '../Loader/Loader';
import RepoCard from '../RepoCard/RepoCard';

import './RepositoriesSearch.scss';

const RepositoriesSearch = () => {
  const [repository, setRepository] = useState('');
  const [data, setData] = useState([]);
  const [loader, setLoader] = useState(false);
  const [errorMessage, setErrorMessage] = useState('');

  const onRepoChange = (e) => setRepository(e.target.value);

  const getRepo = async () => {
    setLoader(true);
    try {
      await Repo.getRepo(repository).then(({ items }) => {
        setData(items);
        setLoader(false);
      });
    } catch (error) {
      console.log(error);
      setErrorMessage(`${error.message}, the repository does not exixt.`);
      setLoader(false);
    }
  };

  const onSearchRepo = (e) => {
    e.preventDefault();
    setErrorMessage('');
    setData([]);
    getRepo();
  };

  return (
    <div className="search__container">
      <Form
        onSubmit={onSearchRepo}
        value={repository}
        onChange={onRepoChange}
      />
      {loader ? (
        <Loader />
      ) : (
        <div className="repo">
          {errorMessage ? (
            <ErrorMessage message={errorMessage} />
          ) : (
            data?.map((repo) => {
              return (
                <RepoCard
                  key={repo.id}
                  name={repo.full_name}
                  url={repo.clone_url}
                  user={repo.owner.login}
                  language={repo.language}
                  fork={repo.forks}
                  stars={repo.stargazers_count}
                  description={repo.description}
                />
              );
            })
          )}
        </div>
      )}
    </div>
  );
};

export default RepositoriesSearch;
