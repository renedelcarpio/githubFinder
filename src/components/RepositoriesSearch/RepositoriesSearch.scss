.repo {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  width: 80%;
}

@media (max-width: 770px) {
  .repo {
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 95%;
  }
}
