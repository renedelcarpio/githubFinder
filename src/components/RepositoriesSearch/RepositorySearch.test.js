import { shallow } from 'enzyme';
import RepositoriesSearch from './RepositoriesSearch';

const wrapper = shallow(<RepositoriesSearch />);
const onSearchRepo = jest.fn();

describe('tests in <RepositorySearch/> component', () => {
  test('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
