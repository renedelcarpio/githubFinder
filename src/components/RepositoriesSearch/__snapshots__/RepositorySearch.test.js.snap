// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in <RepositorySearch/> component should match the snapshot 1`] = `
<div
  className="search__container"
>
  <Form
    onChange={[Function]}
    onSubmit={[Function]}
    value=""
  />
  <div
    className="repo"
  />
</div>
`;
